llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -366,26 +352,38 @@ struct Decomposition {
   Decomposition(int64_t Offset, ArrayRef<DecompEntry> Vars)
       : Offset(Offset), Vars(Vars) {}
 
-  void add(int64_t OtherOffset) {
-    Offset = addWithOverflow(Offset, OtherOffset);
+  // Return true if the new decomposition is invalid.
```

Comment:
```suggestion
  /// Add \p OtherOffset and return true if the operation overflows, i.e. the new decomposition is invalid.
```

would be good to turn them all into doc-dcomments